dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/ToolMetadata.java

```diff
@@ -108,20 +111,42 @@ public static ToolMetadata createFrom(Tool annotation, Bean<?> bean, AnnotatedMe
         SchemaRegistry sr = SchemaRegistry.get();
 
         JsonObject inputSchema = sr.getToolInputSchema(method);
+        boolean isString = unwrappedOutputType.equals(String.class);
+        boolean isContent = unwrappedOutputType.equals(Content.class);
 
         boolean hasContentListReturn = (returnType instanceof ParameterizedType pt && ((Class<?>) pt.getRawType()).isAssignableFrom(List.class)
                                         && (pt.getActualTypeArguments()[0] instanceof Class<?>) && ((Class<?>) pt.getActualTypeArguments()[0]).isAssignableFrom(Content.class));
         boolean hasOutputSchema = (!returnTypeClass.isAssignableFrom(ToolResponse.class) && !hasContentListReturn && !returnTypeClass.isAssignableFrom(Content.class)
                                    && !returnTypeClass.isAssignableFrom(String.class) && annotation.structuredContent());
 
+        if (annotation.structuredContent()) {
+            // Skip schema for basic string/content types
+            if (isString || isContent || hasContentListReturn) {
+                hasOutputSchema = false;
+            } else if (returnTypeClass.equals(ToolResponse.class)) {
+                // Only generate schema if @Schema explicitly provided
+                hasOutputSchema = method.isAnnotationPresent(Schema.class) &&
+                                  !method.getAnnotation(Schema.class).value().equals(Schema.UNSET);
+            } else if (returnType instanceof ParameterizedType pt &&
+                       CompletionStage.class.isAssignableFrom((Class<?>) pt.getRawType())) {
+
+                Type inner = pt.getActualTypeArguments()[0];
+                unwrappedOutputType = resolveOutputType(inner);
+                hasOutputSchema = true;
+            } else {
+                // if it's not a String/Content, we want schema
+                hasOutputSchema = true;
+            }
+        }
```

Comment:
I think we're checking a lot of the same things twice here. Let's go through this code tomorrow.